src/jit/morph.cpp

```diff
@@ -7109,19 +7109,39 @@ GenTreePtr          Compiler::fgMorphCall(GenTreeCall* call)
         else if ((stmtExpr->gtOper == GT_ASG) && (fgMorphStmt->gtNext != nullptr))
         {
             GenTreePtr nextStmtExpr = fgMorphStmt->gtNext->gtStmt.gtStmtExpr;
-            noway_assert(nextStmtExpr->gtOper == GT_RETURN);
+            noway_assert(nextStmtExpr->gtOper == GT_RETURN ||
+                         // There could be a GT_COMMA generated between the call and the return.
+                         // This happens when the result of the call is not used and discarted (in IL ) by a pop operation.
+                         // The generation of the pop IL instruction happens in "case CEE_POP:" in importer.cpp. The GT_COMMA
+                         // is generated by the  gtUnusedValNode(op1) call and it is intended to help the optimizers not to 
+                         // chage the call and it's result in case of the call having side effects or the compilation happens
+                         // for Debug configuration. Handle the potential existance of comma by checking that its second 
+                         // operand is NOP and the next statement is a GT_RETURN.
+                         (nextStmtExpr->gtOper == GT_COMMA && nextStmtExpr->gtOp.gtOp2->gtOper == GT_NOP &&
+                             fgMorphStmt->gtNext->gtNext->gtStmt.gtStmtExpr->OperGet() == GT_RETURN));
```

Comment:
Rather than being duplicated, shouldn't this large expression be factored into a function?